*** Starting uWSGI 2.0.14 (64bit) on [Mon May 21 10:33:49 2018] ***
compiled with version: 4.8.5 20150623 (Red Hat 4.8.5-11) on 11 January 2017 19:48:15
os: Linux-3.10.0-514.el7.x86_64 #1 SMP Tue Nov 22 16:42:41 UTC 2016
nodename: cloud003
machine: x86_64
clock source: unix
pcre jit disabled
detected number of CPU cores: 4
current working directory: /data/guangyong/workspace/work/test/information_push
writing pidfile to uwsgi.pid
detected binary path: /usr/bin/uwsgi
chdir() to /data/guangyong/workspace/work/test/information_push
your processes number limit is 4096
your memory page size is 4096 bytes
detected max file descriptor number: 10000000
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
probably another instance of uWSGI is running on the same address (127.0.0.1:9397).
bind(): Address already in use [core/socket.c line 769]
*** Starting uWSGI 2.0.14 (64bit) on [Mon May 21 14:42:51 2018] ***
compiled with version: 4.8.5 20150623 (Red Hat 4.8.5-11) on 11 January 2017 19:48:15
os: Linux-3.10.0-514.el7.x86_64 #1 SMP Tue Nov 22 16:42:41 UTC 2016
nodename: cloud003
machine: x86_64
clock source: unix
pcre jit disabled
detected number of CPU cores: 4
current working directory: /data/guangyong/workspace/work/information_push
writing pidfile to uwsgi.pid
detected binary path: /usr/bin/uwsgi
chdir() to /data/guangyong/workspace/work/information_push
your processes number limit is 4096
your memory page size is 4096 bytes
detected max file descriptor number: 10000000
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:9396 fd 3
Python version: 2.7.5 (default, Aug  4 2017, 00:39:18)  [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
Python main interpreter initialized at 0x2552100
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 415360 bytes (405 KB) for 8 cores
*** Operational MODE: preforking+threaded ***
unable to load app 0 (mountpoint='') (callable not found or import error)
WSGI app 0 (mountpoint='') ready in 0 seconds on interpreter 0x2552100 pid: 15761 (default app)
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 15761)
spawned uWSGI worker 1 (pid: 15766, cores: 2)
spawned uWSGI worker 2 (pid: 15767, cores: 2)
spawned uWSGI worker 3 (pid: 15769, cores: 2)
spawned uWSGI worker 4 (pid: 15771, cores: 2)
[pid: 15769|app: 0|req: 1/1] 172.16.10.153 () {42 vars in 978 bytes} [Mon May 21 06:42:56 2018] GET /equipment/device/info/?app_id=Fpof8EsZP6i6WfvnhTpMFyfROQMa&device_id=2db04791-f73d-4368-b3e7-0677b731dae5&Token=9f6dd3fd162c23368ef3f6728f1d807a => generated 70 bytes in 562 msecs (HTTP/1.1 200) 3 headers in 119 bytes (1 switches on core 0)
Not Found: /favicon.ico
[pid: 15769|app: 0|req: 2/2] 172.16.10.153 () {44 vars in 840 bytes} [Mon May 21 06:42:56 2018] GET /favicon.ico => generated 2182 bytes in 3 msecs (HTTP/1.1 404) 1 headers in 51 bytes (1 switches on core 1)
[pid: 15769|app: 0|req: 3/3] 172.16.10.153 () {42 vars in 846 bytes} [Mon May 21 06:45:40 2018] GET /equipment/device/info/?app_id=Fpof8EsZP6i6WfvnhTpMFyfROQMa&device_id=2db04791-f73d-4368-b3e7-0677b731dae5&Token=9f6dd3fd162c23368ef3f6728f1d807a => generated 70 bytes in 151 msecs (HTTP/1.1 200) 3 headers in 119 bytes (1 switches on core 0)
[pid: 15767|app: 0|req: 1/4] 172.16.10.153 () {42 vars in 844 bytes} [Mon May 21 06:46:08 2018] GET /equipment/device/info/?app_id=Fpof8EsZP6i6WfvnhTpMFyfROQMa&device_id=2db04791-f73d-4368-b3e7-0677b731dae5&Token=7e278967dfc42285a87624fea5af776 => generated 89 bytes in 538 msecs (HTTP/1.1 200) 3 headers in 119 bytes (1 switches on core 0)
Internal Server Error: /equipment/authentication/auth/
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/lib64/python2.7/site-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib64/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/rest_framework/views.py", line 483, in dispatch
    response = self.handle_exception(exc)
  File "/usr/lib/python2.7/site-packages/rest_framework/views.py", line 443, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/lib/python2.7/site-packages/rest_framework/views.py", line 480, in dispatch
    response = handler(request, *args, **kwargs)
  File "./equipment/authentication/auth_views.py", line 18, in post
    app_id = json.loads(request.body).get("app_id")
  File "/usr/lib64/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
[pid: 15766|app: 0|req: 1/5] 172.16.10.153 () {42 vars in 592 bytes} [Mon May 21 07:01:09 2018] POST /equipment/authentication/auth/ => generated 12840 bytes in 400 msecs (HTTP/1.1 500) 1 headers in 64 bytes (1 switches on core 0)
[pid: 15766|app: 0|req: 2/6] 172.16.10.153 () {44 vars in 647 bytes} [Mon May 21 07:03:54 2018] POST /equipment/authentication/auth/ => generated 153 bytes in 170 msecs (HTTP/1.1 200) 3 headers in 95 bytes (1 switches on core 1)
[pid: 15771|app: 0|req: 1/7] 172.16.10.153 () {44 vars in 872 bytes} [Mon May 21 07:18:05 2018] POST /equipment/device/info/?app_id=Fpof8EsZP6i6WfvnhTpMFyfROQMa&device_id=47914f6b-075c-419e-a5bb-2e87fc753663&Token=a7ebe73e525215f54727f67bfd0a993 => generated 84 bytes in 539 msecs (HTTP/1.1 200) 3 headers in 119 bytes (1 switches on core 0)
[pid: 15766|app: 0|req: 3/8] 172.16.10.153 () {44 vars in 872 bytes} [Mon May 21 07:18:22 2018] POST /equipment/device/info/?app_id=Fpof8EsZP6i6WfvnhTpMFyfROQMa&device_id=47914f6b-075c-419e-a5bb-2e87fc753663&Token=879be5202ed1fd3b72314f918256f52 => generated 118 bytes in 158 msecs (HTTP/1.1 200) 3 headers in 119 bytes (1 switches on core 0)
[pid: 15771|app: 0|req: 2/9] 172.16.10.153 () {44 vars in 876 bytes} [Mon May 21 07:19:20 2018] POST /equipment/commandy/info/?app_id=Fpof8EsZP6i6WfvnhTpMFyfROQMa&device_id=47914f6b-075c-419e-a5bb-2e87fc753663&Token=879be5202ed1fd3b72314f918256f52 => generated 354 bytes in 175 msecs (HTTP/1.1 200) 3 headers in 95 bytes (1 switches on core 1)
[pid: 15766|app: 0|req: 4/10] 172.16.10.153 () {44 vars in 876 bytes} [Mon May 21 07:22:22 2018] POST /equipment/commandy/info/?app_id=Fpof8EsZP6i6WfvnhTpMFyfROQMa&device_id=47914f6b-075c-419e-a5bb-2e87fc753663&Token=879be5202ed1fd3b72314f918256f52 => generated 351 bytes in 159 msecs (HTTP/1.1 200) 3 headers in 95 bytes (1 switches on core 1)
[pid: 15771|app: 0|req: 3/11] 172.16.10.153 () {44 vars in 876 bytes} [Mon May 21 07:26:31 2018] POST /equipment/commandy/info/?app_id=Fpof8EsZP6i6WfvnhTpMFyfROQMa&device_id=47914f6b-075c-419e-a5bb-2e87fc753663&Token=879be5202ed1fd3b72314f918256f52 => generated 351 bytes in 170 msecs (HTTP/1.1 200) 3 headers in 95 bytes (1 switches on core 0)
[pid: 15769|app: 0|req: 4/12] 172.16.10.153 () {44 vars in 876 bytes} [Mon May 21 07:27:37 2018] POST /equipment/commandy/info/?app_id=Fpof8EsZP6i6WfvnhTpMFyfROQMa&device_id=47914f6b-075c-419e-a5bb-2e87fc753663&Token=879be5202ed1fd3b72314f918256f52 => generated 351 bytes in 327 msecs (HTTP/1.1 200) 3 headers in 95 bytes (1 switches on core 1)
[pid: 15766|app: 0|req: 5/13] 172.16.10.153 () {44 vars in 876 bytes} [Mon May 21 07:32:08 2018] POST /equipment/commandy/info/?app_id=Fpof8EsZP6i6WfvnhTpMFyfROQMa&device_id=47914f6b-075c-419e-a5bb-2e87fc753663&Token=879be5202ed1fd3b72314f918256f52 => generated 351 bytes in 160 msecs (HTTP/1.1 200) 3 headers in 95 bytes (1 switches on core 0)
[pid: 15771|app: 0|req: 4/14] 172.16.10.153 () {44 vars in 876 bytes} [Mon May 21 07:38:59 2018] POST /equipment/commandy/info/?app_id=Fpof8EsZP6i6WfvnhTpMFyfROQMa&device_id=2db04791-f73d-4368-b3e7-0677b731dae5&Token=879be5202ed1fd3b72314f918256f52 => generated 352 bytes in 166 msecs (HTTP/1.1 200) 3 headers in 95 bytes (1 switches on core 1)
